### ISSUE TITLE
TreelitePredictorQueryResultSize for multi:softmax gives 10

### ISSUE THREAD

**Issue by apivovarov** [dmlc/treelite#351]
I use C API to get output size of my model  - function `TreelitePredictorQueryResultSize`

The model is mnist digit classifier - script to prepare the model is [here](https://gist.github.com/apivovarov/c1399a2657e575e5253821b4b2482a6b)

It uses the following xgboost train parameters
```
    'objective': 'multi:softmax',   # multiclass classification using the softmax objective
    'num_class': 10                 # the number of classes that exist in this datset
```

I run the model with xgboost - it outputs single label.
I compiled and run the model with treelite python API -  it works fine - it also outputs single label.

My problem is that treelite C API `TreelitePredictorQueryResultSize` gives me 10. But I expect to get 1.

Since treelite python runtime output is correct - only 1 value. I guess I'm doing smth wrong when using C API to get output size.

```
#include <treelite/c_api_runtime.h>
#include <vector>

int main() {
  PredictorHandle treelite_model_;
  int res = TreelitePredictorLoad("./mymodel.so", 1, &treelite_model_);
  printf("TreelitePredictorLoad code: %d\n", res);
  size_t treelite_num_feature_;
  size_t treelite_output_size_;
  size_t num_output_class;
  TreelitePredictorQueryNumFeature(treelite_model_, &treelite_num_feature_);
  TreelitePredictorQueryNumClass(treelite_model_, &num_output_class);

  DMatrixHandle tmp_matrix;
  std::vector<float> tmp_in(treelite_num_feature_);
  const float missing_value = 0.0f;
  TreeliteDMatrixCreateFromMat(tmp_in.data(), "float32", /*num_row=*/1,
                               treelite_num_feature_, &missing_value, &tmp_matrix);

  TreelitePredictorQueryResultSize(treelite_model_, tmp_matrix, &treelite_output_size_);

  const char* predictor_transform;
  TreelitePredictorQueryPredTransform(treelite_model_, &predictor_transform);

  printf("treelite_num_feature_: %d\n", treelite_num_feature_);
  printf("num_output_class: %d\n", num_output_class);
  printf("treelite_output_size_: %d\n", treelite_output_size_);
  printf("predictor_transform: %s\n", predictor_transform);
  return 0;
}
```

Output:
```
treelite_num_feature_: 784
num_output_class: 10
treelite_output_size_: 10
predictor_transform: max_index
```

I also checked `out_result_size` returned by `TreelitePredictorPredictBatch` - it is 1.

**Comment by hcho3** [comment]
This behavior is expected. If you set `pred_margin=1` in `TreelitePredictorPredictBatch`, it will return an array of size 10. This means that `max_index` transformer reduces 10 margin scores into a single float. The predict function will need to have all 10 elements allocated in the result array to write the 10 margin scores; otherwise, you'd get a segfault.
